rust/core/src/automatic_branching.rs

```
@@ -102,6 +110,26 @@ fn hash_source(source: &Source) -> String {
     HEXLOWER.encode(finalised.as_ref())
 }
 
+/// snapshots need to be hashed by their file definiton and yaml config metadata (unique key & strategy)
+fn hash_snapshot(snapshot: &Snapshot) -> String {
+    let mut hasher = Sha256::new();
+    hasher.update(snapshot.file_sha256_hash.as_bytes());
+    hasher.update(snapshot.unique_key.as_bytes());
+    
+    let strategy_bytes = match &snapshot.strategy {
+        Some(strategy) => match &strategy.strategy_type {
+            Some(quary_proto::snapshot::snapshot_strategy::StrategyType::Timestamp(ts)) => ts.updated_at.as_bytes(),
+            None => &[],
+        },
+        None => &[],
+    };
```

"None" or "no strategy" are error cases I don't think should fail silently.